File: src/main/java/com/eternalcode/plots/user/UserRepository.java
package com.eternalcode.plots.user;

import panda.std.reactive.Completable;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public interface UserRepository {

    Completable<List<User>> getUsers();

    CompletableFuture<User> getUser(UUID uuid);

    void saveUser(User user);

}

File: src/main/java/com/eternalcode/plots/user/UserRepository.kt
package com.eternalcode.plots.user

import panda.std.reactive.Completable
import java.util.*
import java.util.concurrent.CompletableFuture

interface UserRepository {
    val users: Completable<List<User?>?>?

    fun getUser(uuid: UUID?): CompletableFuture<User?>?

    fun saveUser(user: User?)
}
